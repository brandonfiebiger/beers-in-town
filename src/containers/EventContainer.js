import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactHtmlParser, { processNodes, convertNodeToElement, htmlparser2 } from 'react-html-parser';

export class EventContainer extends Component {
  constructor(props) {
    super(props);

  }

  displayEvents = () => this.props.events.map(event => {
    return <div>{ ReactHtmlParser(event.desc) }</div>
  })

  render() {
    return(
      <section>
      {this.displayEvents()}
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  events: state.events
})



export default connect(mapStateToProps, null)(EventContainer);