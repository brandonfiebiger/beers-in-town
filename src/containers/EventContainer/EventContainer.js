import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactHtmlParser, { processNodes, convertNodeToElement, htmlparser2 } from 'react-html-parser';
import EventCard from '../EventCard/EventCard'

export class EventContainer extends Component {
  constructor(props) {
    super(props);

  }

  // displayEvents = () => this.props.events.map(event => {
  //   return <div>{ ReactHtmlParser(event.desc) }</div>
  // })

  displayEvents = () => this.props.events.map(event => {
    return <EventCard {...event} />
  })

  render() {
    return(
      <section>
      {this.displayEvents()}
      </section>
    )
  }
}

export const mapStateToProps = state => ({
  events: state.events
})



export default connect(mapStateToProps, null)(EventContainer);