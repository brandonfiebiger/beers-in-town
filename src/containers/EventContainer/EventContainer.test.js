import { EventContainer } from './EventContainer';
import { shallow, mount } from 'enzyme';
import React from 'react';

describe('EventContainer', () => {

  let wrapper;

  beforeEach(() => {

    wrapper = shallow(<EventContainer events={[]}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with events', () => {
    wrapper = shallow(<EventContainer events={[{name: 'Boo'}]}/>)

    expect(wrapper).toMatchSnapshot()

  })
})