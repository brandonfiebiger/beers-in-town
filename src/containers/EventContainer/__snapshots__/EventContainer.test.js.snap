// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventContainer should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventContainer
    events={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "EventContainer",
    },
    "ref": null,
    "rendered": Array [],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "EventContainer",
      },
      "ref": null,
      "rendered": Array [],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`EventContainer should match the snapshot with events 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EventContainer
    events={
        Array [
            Object {
              "name": "Boo",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withRouter(Connect(EventCard))
          name="Boo"
/>,
      ],
      "className": "EventContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "Boo",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Connect(EventCard))
            name="Boo"
/>,
        ],
        "className": "EventContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "Boo",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
